concent_api/api-test.py

```diff
@@ -1,72 +1,152 @@
 #!/usr/bin/env python3
 
 import sys
-import json
 import datetime
 import random
+import http.client
+from base64                             import b64encode
+
+from golem_messages.message             import Message, MessageForceReportComputedTask, MessageRejectReportComputedTask, MessageCannotComputeTask, MessageTaskToCompute, MessageAckReportComputedTask
+from golem_messages.shortcuts           import dump, load
 
 import requests
 
+CONCENT_PUBLIC_KEY  = b'\xf3\x97\x19\xcdX\xda\x86tiP\x1c&\xd39M\x9e\xa4\xddb\x89\xb5,]O\xd5cR\x84\xb85\xed\xc9\xa17e,\xb2s\xeb\n1\xcaN.l\xba\xc3\xb7\xc2\xba\xff\xabN\xde\xb3\x0b\xa6l\xbf6o\x81\xe0;'
+CONCENT_PRIVATE_KEY = b'l\xcdh\x19\xeb$>\xbcG\xa1\xc7v\xe8\xd7o\x0c\xbf\x0e\x0fM\x89lw\x1e\xd7K\xd6Hnv$\xa2'
 
 DEFAULT_HEADERS = {
-    'Content-Type':              'application/json',
-    'concent-client-public-key': '85cZzVjahnRpUBwm0zlNnqTdYom1LF1P1WNShLg17cmhN2UssnPrCjHKTi5susO3wrr/q07eswumbL82b4HgOw==',
+    'Content-Type':              'application/octet-stream',
+    'concent-client-public-key': b64encode(CONCENT_PUBLIC_KEY).decode('ascii'),
 }
 
 
 def create_data(message_type, task_id):
+    assert task_id > 0
+
     current_time = int(datetime.datetime.now().timestamp())
-    data = {
-        "type":         message_type,
-        "timestamp":    current_time,
-        "message_task_to_compute": {
-            "type":         "MessageTaskToCompute",
-            "timestamp":    current_time,
-            "task_id":      task_id,
-            "deadline":     current_time + 6000
-        }
-    }
-    return data
-
-
-def create_reject_data(message_type, task_id):
+    message_task_to_compute = MessageTaskToCompute(
+        timestamp = current_time,
+        task_id = task_id,
+        deadline = current_time + 6000,
+    )
+
+    message_task_to_compute_dumped = dump(
+        message_task_to_compute,
+        CONCENT_PRIVATE_KEY,
+        CONCENT_PUBLIC_KEY,
+    )
+
+    if message_type == 'MessageForceReportComputedTask':
+        return MessageForceReportComputedTask(
+            timestamp = current_time,
+            message_task_to_compute = message_task_to_compute_dumped
+        )
+
+    elif message_type == 'MessageAckReportComputedTask':
+        return MessageAckReportComputedTask(
+            timestamp = current_time,
+            message_task_to_compute = message_task_to_compute_dumped,
+        )
+    else:
+        assert False, 'Invalid message type'
+
+    return None
```

Comment:
return None is not needed, this is what function returns by default when no return is specified.